### DISCUSSION TITLE
is `twitter:card` expected even when twitterId not set and no image?

### DISCUSSION THREAD

**Discussion by narumi147** [vuepress-theme-hope/vuepress-theme-hope#1587] [Q&A] [answered]
Page settings: no setting, default md file, text only, without image.

```ts
  [ 'meta', { name: 'twitter:card', content: 'summary_large_image' } ],
  [ 'meta', { name: 'twitter:image:alt', content: 'Chaldea' } ]
```
These two are always added even twitterId not set and without image. And `og:image` is also not added.
So I think it's better not to append twitter card tag when both are empty.
`og:description` is not added too, not sure where you parse this from. 

Furthermore, when there is no image on this page, It can be fallback to site icon, and description fallback to `<meta name="description">`

**Comment by Mister-Hope** [comment]
Nice catch

**Reply by narumi147** [reply]
Actually it broke the discord default link embed behaviour because of `twitter:card=summary_large_image`. That's why I ask here.
And I have to remove twitter tag manually as temp solution.

**Reply by Mister-Hope** [reply]
I will have a fix soon.

**Comment by Mister-Hope** [comment, marked as answer]
Track https://github.com/vuepress-theme-hope/vuepress-theme-hope/issues/1591
